service/s3/internal/customizations/update_endpoint.go

```diff
@@ -48,24 +83,66 @@ func (u *updateEndpointMiddleware) HandleSerialize(
 		return out, metadata, fmt.Errorf("unknown request type %T", req)
 	}
 
-	// Below customization only apply if bucket name is provided
-	bucket, ok := u.getBucketFromInput(in.Parameters)
-	if ok && bucket != nil {
-		if err := u.updateEndpointFromConfig(req, *bucket); err != nil {
-			return out, metadata, err
+	// check if accelerate is supported
+	if u.supportsAccelerate == nil || (u.useAccelerate && !u.supportsAccelerate(in.Parameters)) {
+		// accelerate is not supported, thus will be ignored
+		log.Println("Transfer acceleration is not supported for the operation, ignoring UseAccelerate.")
+		u.useAccelerate = false
+	}
+
+	// transfer acceleration is not supported with path style urls
+	if u.useAccelerate && u.usePathStyle {
+		log.Println("UseAccelerate is not compatible with UsePathStyle, ignoring UsePathStyle.")
+		u.usePathStyle = false
+	}
+
+	if u.getBucketFromInput != nil {
+		// Below customization only apply if bucket name is provided
+		bucket, ok := u.getBucketFromInput(in.Parameters)
+		if ok && bucket != nil {
+			if err := u.updateEndpointFromConfig(req, *bucket); err != nil {
+				return out, metadata, err
+			}
 		}
 	}
+
 	return next.HandleSerialize(ctx, in)
 }
 
 func (u updateEndpointMiddleware) updateEndpointFromConfig(req *http.Request, bucket string) error {
+	// do nothing if path style is enforced
 	if !u.usePathStyle {
```

Comment:
Could this just return early if value is set to `u.usePathstyle`? instead of the large if block?